md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/HiveTableMigrationAPI.java

```diff
@@ -0,0 +1,369 @@
+                 continue;
+             }
+             if (string.startsWith("srcDB_")) {
+                 dbName = map.get(string);
+             }
+         }
+         LOGGER.info("Database is"+dbName);
+         RestWrapperOptions restWrapperOptions = null;
+         try {
+             DatabaseMetaData metaData = getHiveJDBCConnection(dbName).getMetaData();
+
+             ResultSet rs = metaData.getTables(null, null, null, null);
+             List<String> tables = new ArrayList<String>();
+             while (rs.next()) {
+                 String tableName = rs.getString("TABLE_NAME");
+                 tables.add(tableName.toUpperCase());
+             }
+
+             try {
+                 closeResultset(rs);
+                 closeConnection();
+             } catch (SQLException ex) {
+                 LOGGER.error("Error in close");
+             } finally {
+                 connection = null;
+             }
+             List<RestWrapperOptions.Option> options = new ArrayList<RestWrapperOptions.Option>();
+             for(String table : tables)
+             {
+                 RestWrapperOptions.Option option = new RestWrapperOptions.Option(table,table);
+                 options.add(option);
+             }
+             restWrapperOptions = new RestWrapperOptions(options, RestWrapperOptions.OK);
+         } catch (Exception e) {
+             LOGGER.error("error occured :" + e);
+             restWrapperOptions = new RestWrapperOptions(e.getMessage(), RestWrapperOptions.ERROR);
+         }
+         return restWrapperOptions;
+     }
+
+ */
+    @RequestMapping(value = "/createjobs/{checked}", method = RequestMethod.POST)
+    @ResponseBody
+    public RestWrapper createJob(@RequestParam Map<String, String> map, Principal principal, @PathVariable("checked") String[] checkedTables) {
+        LOGGER.debug(" value of map is " + map.size());
+        RestWrapper restWrapper = null;
+
+        String processName = null;
+        String processDesc = null;
+        Integer busDomainID = null;
+        Integer tablesSize = 0;
+        String nameNodeIp = map.get("scrNameNode");
+        String jobTrackerIp = map.get("srcJobTracker");
+        String destnameNodeIp = map.get("destNameNode");
+        String destjobTrackerIp = map.get("destjobTracker");
+        com.wipro.ats.bdre.md.dao.jpa.Properties jpaProperties = null;
+        for (int i = 1; i <= checkedTables.length; i++)
+            LOGGER.info("table is " + checkedTables[i - 1]);
+        List<com.wipro.ats.bdre.md.beans.table.Process> allTableProcessList = new ArrayList<com.wipro.ats.bdre.md.beans.table.Process>();
+
+        for (int i = 1; i <= checkedTables.length; i++) {
+            List<Properties> propertiesList = new ArrayList<Properties>();
+            LOGGER.info("table name is " + checkedTables[i - 1]);
+            jpaProperties = Dao2TableUtil.buildJPAProperties("hive-migration", "src-nn", nameNodeIp, "SourceNameNodeAddress");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "hive-migration" 10 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1192)